packages/mui-system/src/spacing/spacing.js

```diff
@@ -93,30 +93,35 @@ export const paddingKeys = [
 const spacingKeys = [...marginKeys, ...paddingKeys];
 
 export function createUnaryUnit(theme, themeKey, defaultValue, propName) {
-  const themeSpacing = getPath(theme, themeKey, false) ?? defaultValue;
+  const themeSpacing = getPath(theme, themeKey, true) ?? defaultValue;
 
-  if (typeof themeSpacing === 'number') {
-    return (abs) => {
-      if (typeof abs === 'string') {
-        return abs;
+  if (typeof themeSpacing === 'number' || typeof themeSpacing === 'string') {
+    return (val) => {
```

Comment:
I changed the variable name to reflect what we expect because the `val` is no longer an absolute of the user input. It's the user input without any modification.